### PR TITLE
feat(scaffolder): Add git author for gitHub pull request scaffolder action

### PR DESCRIPTION
**Pull request by legendik**
## Hey, I just made a Pull Request!

Closes https://github.com/backstage/backstage/issues/17644, heavily inspired by PR https://github.com/backstage/backstage/pull/17827 and comments in the PR :)

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/scaffolder-backend/src/scaffolder/actions/builtin/publish/githubPullRequest.ts [hunk 1]
```diff
@@ -312,6 +332,21 @@ export const createPublishGithubPullRequestAction = (
...
+          name: gitAuthorName
+            ? gitAuthorName
+            : config.getOptionalString('scaffolder.defaultAuthor.name'),
+          email: gitAuthorEmail
+            ? gitAuthorEmail
+            : config.getOptionalString('scaffolder.defaultAuthor.email'),
+        };
+
+        // use provided info if possible, otherwise use fallbacks
+        const authorInfo = {
```
Comment (benjdlambert) [inline plugins/scaffolder-backend/src/scaffolder/actions/builtin/publish/githubPullRequest.ts:345]: I wonder if we can use the token that we're going to use here to set some sane defaults. Providing we can get name and email from the token and the `/user` endpoint?

**Review by benjdlambert — COMMENTED** [review]
Hey - just a question here about the defaultS?

**Comment by legendik** [comment]
> Hey - just a question here about the defaultS?

Hi, i was inspired by github publish action, where the defaults are set https://github.com/backstage/backstage/blob/master/plugins/scaffolder-backend/src/scaffolder/actions/builtin/helpers.ts#L163

File: plugins/scaffolder-backend/src/scaffolder/actions/builtin/publish/githubPullRequest.ts (hunk 1 above)
Comment (legendik) [reply plugins/scaffolder-backend/src/scaffolder/actions/builtin/publish/githubPullRequest.ts:345]: Hi, i was inspired by github publish action, where the defaults are set https://github.com/backstage/backstage/blob/master/plugins/scaffolder-backend/src/scaffolder/actions/builtin/helpers.ts#L163

Comment (benjdlambert) [reply plugins/scaffolder-backend/src/scaffolder/actions/builtin/publish/githubPullRequest.ts:345]: Yea - I think there's also room to update that there too. My problem with this change is that we didn't provide a `committer` before, what is the commiter set as? Is it the signed in users token details? In that case then we probably want to keep that backwards compatibility by not adding one instead?

Comment (benjdlambert) [reply plugins/scaffolder-backend/src/scaffolder/actions/builtin/publish/githubPullRequest.ts:345]: @legendik just a ping here ^

**Pull request closed without merging**